src/ghci/stdout.rs

```diff
@@ -99,31 +116,24 @@ impl GhciStdout {
             .reader
             .read_until(&bootup_patterns, WriteBehavior::Write, &mut self.buffer)
             .await?;
-        tracing::debug!(?lines, "ghci started");
+        tracing::debug!(?lines, "ghci started, saw version marker");
 
-        let init_prompt_patterns = AhoCorasick::from_anchored_patterns(["ghci> "]);
         // We know that we'll get _one_ `ghci> ` prompt on startup.
-        self.prompt(when_ready, Some(&init_prompt_patterns)).await?;
-        // But we might get more than one `ghci> ` prompt, so digest any others from the buffer.
-        while let Some(lines) = self
-            .reader
-            .try_read_until(
-                &init_prompt_patterns,
-                WriteBehavior::NoFinalLine,
-                &mut self.buffer,
-            )
-            .await?
-        {
-            tracing::debug!(?lines, "initial ghci prompt");
-        }
+        let init_prompt_patterns = AhoCorasick::from_anchored_patterns(["ghci> "]);
+        let (sender, receiver) = oneshot::channel();
```

Comment:
What is the purpose of this extra channel?  The reason I ask is because, looking at the `prompt` method's definition it seems like the `sender` channel here isn't passed off to any other thread and is only used to receive the result of the `prompt` function, but then I'm wondering if the `prompt` could just return that result instead of passing it back via this channel?

Reply:
Usually we're calling the `prompt` function from a different thread; the general flow is the `GhciStdin` job writes a line to stdin and then sends a `Prompt` message to this channel, which waits until it sees a prompt on stdout, and then notifies the sender that it saw a prompt. (This is because sending a message to `GhciStdout` will usually complete immediately, but the prompt will only appear later, so we need a synchronization mechanism.)

For this particular call, we could just call the method, but it needs the `sender` parameter for the majority of use-cases.

This channel also isn't entirely new, it was just moved from the `Ghci::new` method in `src/ghci/mod.rs` to here.